variables.tf

```diff
@@ -743,6 +743,22 @@ variable "efs_file_system_token" {
   default     = ""
 }
 
+variable "efs_throughput_mode" {
+  description = "(Optional) Throughput mode for the file system. Defaults to bursting. Valid values: bursting, provisioned, or elastic. When using provisioned, also set provisioned_throughput_in_mibps."
+  type        = string
+  default     = "bursting"
+  validation {
+    condition     = contains(["bursting", "provisioned", "elastic"], var.efs_throughput_mode)
+    error_message = "The mode must be one of \"bursting\" \"provisioned\" or \"elastic\""
+  }
+}
+
+variable "efs_provisioned_throughput_in_mibps" {
+  description = "The throughput, measured in MiB/s, that you want to provision for the file system. Only applicable with efs_throughput_mode set to provisioned"
+  type        = number
+  default     = null
```

Comment:
What happens if this is null and set to provisioned.  Is it worth adding validation to ensure that it is a valid number if efs_througpuhs_mode is set to provisioned?

Reply:
You cannot do validation on another variable within a single variable. I did try adding validation for exactly this and terraform validate failed with an error indicating this wasn't possible.